src/renderers/webgl/WebGLProgram.js

```diff
@@ -25,20 +30,58 @@ THREE.WebGLProgram = ( function () {
 		var uniforms = {};
 
 		var n = gl.getProgramParameter( program, gl.ACTIVE_UNIFORMS );
-
+	
 		for ( var i = 0; i < n; i ++ ) {
 
 			var info = gl.getActiveUniform( program, i );
 			var name = info.name;
 			var location = gl.getUniformLocation( program, name );
 
-			// console.log("THREE.WebGLProgram: ACTIVE UNIFORM:", name);
+			//console.log("THREE.WebGLProgram: ACTIVE UNIFORM:", name);
+
+			var matches = structRe.exec(name);
+			if( matches ) {
+
+				var structName = matches[1];
+				var structProperty = matches[2];
+
+				var uniformsStruct = uniforms[ structName ];
+				if( ! uniformsStruct ) {
+					uniformsStruct = uniforms[ structName ] = {};
+				}
+				uniformsStruct[ structProperty ] = location;
+
+				continue;
+			}
+
+			matches = arrayStructRe.exec(name);
+			if( matches ) {
```

Comment:
Yes, I in fact I have been already been inspired by this. Missing `ShaderMaterial.toJSON` & Loader, isn't entirely unrelated, and once we can do this stuff with structures and everything, it would be a shame to only use it for uniforms....

However, I stopped at some point, as a matter of sanity. 

> I [ ... ] can not consistently contribute to ThreeJS unless my work objectives are well aligned with my ThreeJS contributions.

You are not the only one. Currently I need to get the animation stuff ready, plus basic authoring.